lib/srv/db/mongodb/protocol/message.go

```diff
@@ -101,8 +102,8 @@ func readHeaderAndPayload(reader io.Reader) (*MessageHeader, []byte, error) {
 		return nil, nil, trace.BadParameter("invalid header size %v", header)
 	}
 
-	payloadLength := int64(length - headerSizeBytes)
-	if payloadLength >= defaultMaxMessageSizeBytes {
+	payloadLength := uint32(length - headerSizeBytes)
+	if payloadLength >= maxMessageSize {
```

Comment:
why the change from 64bit to 32?

Reply:
I couldn't find documentation mentioning the field type, so I'm assuming [what the driver uses](https://github.com/mongodb/mongo-go-driver/blob/59f7519e8b7c4e549a39ce769b8536f2bdfd0446/mongo/description/server.go#L51).